File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/PassResult.java
package com.armongate.mobilepasssdk.model;

public class PassResult {
    public boolean  success;
    public Integer  failCode;
    public Integer  direction;
    public String   clubId;
    public String   clubName;

    public PassResult(boolean success, Integer direction, String clubId, String clubName, Integer failCode) {
        this.success    = success;
        this.direction  = direction;
        this.clubId     = clubId;
        this.clubName   = clubName;
        this.failCode   = failCode;
    }
}

File: ios/MobilePassSDK/Model/PassResult.swift
//
//  PassResult.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 26.07.2022.
//

import Foundation

public struct PassResult {
    public var success:     Bool
    public var failCode:    Int?
    public var direction:   Int?
    public var clubId:      String?
    public var clubName:    String?
    
    init(success: Bool, direction: Direction?, clubId: String?, clubName: String?, failCode: Int?) {
        self.success    = success
        self.direction  = direction != nil ? direction!.rawValue : nil
        self.clubId     = clubId
        self.clubName   = clubName
        self.failCode   = failCode
    }
}
